### DISCUSSION TITLE
[Feat]: Internal Browser

### DISCUSSION THREAD

**Discussion by aghasemi** [coder/code-server#4791] [Q&A]
Hi. Is there a way to open web links inside the Code Server (i.e. in another window in the IDE), rather than the enclosing browser?

As an example use case, heroku login doesn't work the current way because of an "IP Mismatch" (A workaround to this is using interactive login). There are other benefits as well. 

Thanks

**Comment by mauri-sfdc** [comment]
Could you expand on the heroku login issue? where's the IP Mismatch and how could this solve it.

Thanks

**Comment by aghasemi** [comment]
When you run `heroku login`, it gives you a link you should open in your browser to create an OTP and log in the Heroku cli (2FA etc). However, if you do so from the code-server and open the given link in the browser of your local machine, Heroku complains "IP Mismatch", possibly because the machine on which the link is generated is not the machine on which the browser is running.

**Reply by jsjoeio** [reply]
Tools like `vercel` have a special flag for this: https://vercel.com/docs/cli#commands/login/unique-options/out-of-band-mode

Did you try using the [`--interactive` flag](https://devcenter.heroku.com/articles/heroku-cli-commands#heroku-auth-login)?

**Comment by jsjoeio** [comment]
Please use our Q and A for things like this!
